### PR TITLE
User search field

### PR DESCRIPTION
**Pull request by ActiveChooN**
### Motivation and context

изображение

This PR provide user search field for assignee field for the task. It should increase UX with user search by username and increase loading speed for cvat.org.

### How has this been tested?

Manually

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
~~- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly~~
~~- [ ] I have added tests to cover my changes~~
~~- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))~~
- [x] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2020 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by bsekachev** [comment]
Could you please merge develop to CI passes?

**Comment by bsekachev** [comment]
It looks like some unit tests of cvat-core & cypress tests are broken

**Comment by ActiveChooN** [comment]
@bsekachev, all tests was fixed

**Comment by bsekachev** [comment]
Now each new character creates new HTTP request. How about debouncing a user input? 0.5-1 seconds I suppose would be appropriate.
Screenshot from 2020-11-10 12-44-33

**Comment by bsekachev** [comment]
It is not very obvious how to remove assignee with current UI element. To do that I need to remove the text and click outside the input field (enter doesn't work for example). I suppose it might be more user-friendly. 
One more approach which probably better I see is adding X icon at the right of input for the purpose.

**Comment by bsekachev** [comment]
@ActiveChooN 
On other points the patch is nice and it significantly accelerates the user interface. Let's fix left comments, resolve conflicts and merge it

**Comment by bsekachev** [comment]
@dvkruchinin Hi, on CI we get ``Failed to connect to Chrome, retrying in 1 second (attempt 23/62)``. As far as I remember you've fixed something like that. Doesn't the fix work? Maybe try to find another ways or workarounds? It is not a rare issue

File: cvat-core/.eslintrc.js [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 module.exports = {
     env: {
+        amd: true,
```
[the remark this answers is not in the document]
Comment (ActiveChooN) [reply cvat-core/.eslintrc.js:7]: Eslint warn about require usage without that flag.

[the remark this answers is not in the document]
Comment (bsekachev) [reply cvat-core/.eslintrc.js:7]: Just set ``node`` to true

**Comment by dvkruchinin** [comment]
> @dvkruchinin Hi, on CI we get `Failed to connect to Chrome, retrying in 1 second (attempt 23/62)`. As far as I remember you've fixed something like that. Doesn't the fix work? Maybe try to find another ways or workarounds? It is not a rare issue

Hi,
It was more of a test. Since in the discussions, cypress users wrote that someone this fix helped, someone did not. Yes, I'll keep looking. And I want to believe that this error has become less common.

**Comment by ActiveChooN** [comment]
@bsekachev 

> 
> 
> It is not very obvious how to remove assignee with current UI element. To do that I need to remove the text and click outside the input field (enter doesn't work for example). I suppose it might be more user-friendly.
> One more approach which probably better I see is adding X icon at the right of input for the purpose.

Added `Enter` button for empty search request

> I suppose the styles also applicable for the new element

Strange margins here, removed styles

**Review by bsekachev — APPROVED** [review]

**Pull request merged**